effect_prn/models/end_of_study.py

```diff
@@ -25,40 +36,129 @@ class EndOfStudy(
+    offschedule_reason = models.CharField(
+        verbose_name="Reason patient was terminated from the study",
+        choices=STUDY_TERMINATION_REASONS,
+        max_length=50,
+        null=True,
+    )
+
+    offschedule_reason_other = OtherCharField()
+
+    withdrawal_consent_reasons = models.TextField(
+        verbose_name="If withdrawal Consent, please specify reasons",
+        max_length=500,
+        blank=True,
+        null=True,
+    )
+
+    late_exclusion_reasons = models.TextField(
+        verbose_name="If late exclusion for other reason, specify reason",
+        max_length=500,
         blank=True,
         null=True,
     )
 
     transferred_consent = models.CharField(
         verbose_name=(
-            "If transferred, has the patient provided consent to be followed-up?"
+            "If transferred, has the patient provided consent to be followed-up for 6 month end-point?"
         ),
         choices=YES_NO_NA,
         max_length=15,
         default=NOT_APPLICABLE,
     )
 
+    medication_study_termination = models.CharField(
+        verbose_name="Medicines on Study Termination day",
+        choices=MEDICINES,
+        max_length=45,
+    )
+
+    medication_study_termination_other = OtherCharField()
+
+    dx_since_enrolment = models.CharField(
+        verbose_name="Other significant diagnoses since enrolment",
+        choices=YES_NO,
+        max_length=15,
+    )
+
+    dx_since_enrolment_date = models.DateField(
+        verbose_name="If Yes, date of diagnosis",
+        validators=[date_not_future],
+        blank=True,
+        null=True,
+    )
+
+    dx_since_enrolment_dx = models.CharField(
+        verbose_name="If Yes, diagnosis",
+        choices=DIAGNOSIS,
+        max_length=45,
+    )
+
+    dx_since_enrolment_other = OtherCharField()
+
+    missed_doses_5fc_cnt = models.IntegerField(
+        verbose_name="Number of doses missed in first 14 days: 5FC",
+        blank=False,
+        null=True,
+    )
+
+    missed_doses_flu_cnt = models.IntegerField(
+        verbose_name="Number of doses missed in first 14 days: FLU",
+        blank=False,
+        null=True,
+    )
+
+    fcon_dose_14 = models.CharField(
+        verbose_name="Fluconazole dose taken during first 14days of study",
+        choices=FLUCONAZOLE_DOSE_14DAYS,
+        max_length=15,
+    )
+
+    fcon_dose_14_other = OtherCharField()
+
+    fcon_dose_14_reasons = models.TextField(
+        verbose_name="Reasons",
+        max_length=500,
+        blank=True,
+        null=True,
+    )
+
+    missed_doses_consolidation_flu_cnt = models.IntegerField(
+        verbose_name="Total number of doses missed during consolidation/maintenance phase: FLU",
+        blank=True,
+        null=True,
+    )
+
+    fcyz_consolidation_phase = models.CharField(
+        verbose_name="Fluconazole dose taken during consolidation/maintenance phase",
+        choices=FLUCONAZOLE_DOSE_CONSOLIDATION,
+        max_length=15,
+    )
```

Comment:
Should this be fluconazole, or flucytosine (the verbose_name doesn't match the field name)